Automattic/newspack-plugin
# Unified Engagement Wizard

### All Submissions:






### Changes proposed in this Pull Request:

This  is a stub Engagement Wizard, with navigation UI and basic functionality for most features. 

#### Newsletters

<img width="1700" alt="Screen Shot 2019-09-12 at 11 40 51 PM" src="https://user-images.githubusercontent.com/1477002/64836143-c8bb6800-d5b6-11e9-8bc2-c9e1d3a3aab3.png">

This is a combination of  the  existing "Set Up Mailchimp" and "Set Up Newsletter Subscriptions Block" Wizards. The upper `Card` provides description and link to connect a Mailchimp account on WPCOM. The lower `Card` manages installation of the WooCommerce Mailchimp extension and allows input of Mailchimp API key. The lower section will only be shown if WooCommerce is installed and active. Note that the API Key is not persisted, which is the status quo in the "Set Up Mailchimp" Wizard.

#### Social

This page contains an `ActionCard` with Handoff to the Jetpack Publicize feature.

<img width="1603" alt="Screen Shot 2019-09-12 at 11 51 32 PM" src="https://user-images.githubusercontent.com/1477002/64836554-6e230b80-d5b8-11e9-83ef-8bc1b84f2ee5.png">

#### Commenting

Commenting has a three-item submenu. `Disqus` and `The Coral Project` pages will install the respective plugins, and then show an `ActionCard` with handoff to their dashboards. `WordPress Discussion` is an `ActionCard` that simply links to Discussion settings. This isn't a true Handoff because of some complexities linking to a page that isn't associated with an installed plugin. I elected not to resolve this issue yet, since it's possible that instead of sending to the Discussion dashboard this screen will be responsible for configuring it. 

<img width="1603" alt="Screen Shot 2019-09-12 at 11 59 31 PM" src="https://user-images.githubusercontent.com/1477002/64837019-7bd99080-d5ba-11e9-8ff6-f4d40ae48345.png">

#### UGC

The UGC area is placeholder.

<img width="1603" alt="Screen Shot 2019-09-13 at 12 10 27 AM" src="https://user-images.githubusercontent.com/1477002/64837106-eb4f8000-d5ba-11e9-861b-7ff9d06e1af8.png">

This PR also adds a notification (based on https://github.com/Automattic/newspack-plugin/pull/267) if Disqus plugin is active but Newspack AMP Disqus is not. Between the automated installation of the two together and this notification, this should supersede https://github.com/Automattic/newspack-plugin/pull/267.

### How to test the changes in this Pull Request:

1. Checkout the branch and run  `npm run build:webpack`
2. Navigate to Engagement Wizard.
3. Click through all pages, verify that their appearance and behavior matches the descriptions and screenshots above. 

### Other information:
